### PR TITLE
CAPI: Make it possible to create struct types

### PR DESCRIPTION
**Pull request by alnkesq**
The C API was previously able to only create union logical types -- not structs. This PR adds support for structs as well.

Note: I used `duckdb_logical_type *` instead of `duckdb_logical_type` like in `duckdb_create_union_type`, because non-C callers of this function might not know `sizeof(duckdb::LogicalType)` and so might be unable to allocate the `member_types` array. For now, I only changed the signature of this new function.

### CODE REVIEW

File: src/main/capi/logical_types-c.cpp [hunk 1]
```diff
@@ -51,6 +51,22 @@ duckdb_logical_type duckdb_create_union_type(duckdb_logical_type member_types_p,
...
+                                              idx_t member_count) {
+	if (!member_types_p || !member_names) {
+		return nullptr;
+	}
+	duckdb::LogicalType **member_types = (duckdb::LogicalType **)member_types_p;
+	duckdb::LogicalType *mtype = new duckdb::LogicalType;
+	duckdb::child_list_t<duckdb::LogicalType> members;
+
+	for (idx_t i = 0; i < member_count; i++) {
+		members.push_back(make_pair(member_names[i], *member_types[i]));
```
Comment (Tishj) [inline src/main/capi/logical_types-c.cpp:64]: Maybe we want to verify that none of the names are NULL, and return error instead of segfault in that case

**Review by Mytherin — COMMENTED** [review]
Thanks for the PR! LGTM. Could you just add a test to the C API tests that uses this function?

**Comment by alnkesq** [comment]
I added a test.

**Comment by alnkesq** [comment]
@Mytherin The CI failure seems to be unrelated (Python "Extension "/project/httpfs/httpfs.duckdb_extension" version (v..1-dev) does not match DuckDB version (0.0.0)")

**Comment by Mytherin** [comment]
Thanks! LGTM

**Pull request merged**